src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,20 +8,30 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, some street, #09-123, S123456";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String COMMA = ",";
 
     public final String value;
     private boolean isPrivate;
-
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
+    
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        String[] processAddress = trimmedAddress.split(",");
+        block = new Block(processAddress[0].trim());
+        street = new Street(processAddress[1].trim());
+        unit = new Unit(processAddress[2].trim());
+        postalCode = new PostalCode(processAddress[3].trim());
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```

Comment:
How about using the smaller classes for their respective validation to increase precision?